x/ccv/provider/keeper/hooks.go

```diff
@@ -175,7 +175,46 @@ func (h Hooks) AfterProposalVotingPeriodEnded(ctx sdk.Context, proposalID uint64
 func (h Hooks) AfterProposalDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress) {
 }
 
+// AfterProposalVote opts in validators that vote YES (with 100% weight) on a `ConsumerAdditionProposal`. If a
+// validator votes multiple times, only the last vote would be considered on whether the validator is opted in or not.
 func (h Hooks) AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress) {
+	validator, found := h.k.stakingKeeper.GetValidator(ctx, voterAddr.Bytes())
+	if !found {
+		return
+	}
+
+	consAddr, err := validator.GetConsAddr()
+	if err != nil {
+		return
```

Comment:
Note: These are cases where we could also `panic` instead of simply `return`ing. We could `panic` because those are exceptional cases that should never trigger (e.g., a validator with no consensus address). However, I choose to simply `return` because a potential buggy `panic` call would potentially imply that noone could `MsgVote` on such a proposal.